rero_ils/modules/acq_orders/serializers/csv.py

```diff
@@ -0,0 +1,283 @@
+                    document_data = {
+                        'document_title': next(
+                            filter(lambda x: x.get('type') == 'bf:Title',
+                                   data.get('title'))
+                        ).get('_text')
+                    }
+                    # process provision activity
+                    provision_activity = next(
+                        filter(lambda x: x.get('type') == 'bf:Publication',
+                               data.get('provisionActivity'))
+                    )
+                    start_date = provision_activity.get('startDate', '')
+                    end_date = provision_activity.get('endDate')
+                    document_data['document_publication_year'] = \
+                        f'{start_date} - {end_date}' \
+                        if end_date else start_date
+
+                    document_data['document_publisher'] = ' | '.join(
+                        data['value']
+                        for stmt in provision_activity.get('statement', [])
+                        for data in stmt.get('label', [])
+                        if stmt['type'] == 'bf:Agent'
+                    )
+                    document_data['document_edition_statement'] = ' | '.join(
+                        edition.get('value')
+                        for edition_statement in
+                        data.get('editionStatement', [])
+                        for edition in edition_statement.get('_text', [])
+                    )
+                    document_data['document_series_statement'] = ' | '.join(
+                        data['value']
+                        for serie in data.get('seriesStatement', [])
+                        for data in serie.get('_text', [])
+                    )
+                    document_data['document_isbn'] = ' | '.join(
+                        data.get('isbn', [])
+                    )
+
+                    # process contributions
+                    creator = []
+                    if 'contribution' in data:
+                        for contribution in data.get('contribution'):
+                            if any(role in contribution.get('role')
+                                   for role in role_filter):
+                                authorized_access_point = \
+                                    f'authorized_access_point_{language}'
+                                if authorized_access_point in contribution\
+                                        .get('agent'):
+                                    creator.append(
+                                        contribution['agent'][
+                                            authorized_access_point]
+                                    )
+
+                    document_data['document_creator'] = ' ; '.join(creator)
+
+                    return document_data
+
+                records = DocumentsSearch() \
+                    .get_records_by_terms(terms=doc_pids)
```

Comment:
Sure that you need all fields? Can you check also for other resources?

Reply:
Done